lib/puppet/util/execution.rb

```diff
@@ -98,11 +98,11 @@ def self.exitstatus
   # @raise [exception] under same conditions as {execute}, but raises the given `exception` with the output as argument
   # @return (see execute)
   # @api public
+  # @deprecated
```

Comment:
Two other quick notes:

* we agreed that it's fine to skip deprecation messages here for end users - given the inline docs should be enough (noting that `execfail` is an external API given the `@api public`)
* we'll add a ticket for PUP 6 removal